src/components/VModal/VModalContent.vue

```diff
@@ -92,7 +92,7 @@ const VModalContent = defineComponent({
     },
     hideOnClickOutside: {
       type: Boolean,
-      default: true,
+      default: false,
```

Comment:
This is the change that fixes the bug, everything else is ... refactoring, I guess :) and tests